frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/font_family_setting.dart

```diff
@@ -109,35 +111,61 @@ class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
         .replaceAllMapped(camelCase, (m) => ' ${m.group(0)}');
   }
 
+  bool _isTextOverflowing(String text, TextStyle textStyle, double maxWidth) {
+    final textPainter = TextPainter(
+      text: TextSpan(text: text, style: textStyle),
+      maxLines: 1,
+      textDirection: ui.TextDirection.ltr,
+    )..layout(maxWidth: maxWidth);
+
+    return textPainter.didExceedMaxLines;
+  }
+
   Widget _fontFamilyItemButton(BuildContext context, TextStyle style) {
     final buttonFontFamily = parseFontFamilyName(style.fontFamily!);
-    return SizedBox(
-      key: UniqueKey(),
-      height: 32,
-      child: FlowyButton(
-        key: Key(buttonFontFamily),
-        onHover: (_) => FocusScope.of(context).unfocus(),
-        text: FlowyText.medium(
-          parseFontFamilyName(style.fontFamily!),
-          fontFamily: style.fontFamily!,
+
+    final bool isOverFlown = _isTextOverflowing(
+      buttonFontFamily,
+      Theme.of(context).textTheme.bodyMedium!.copyWith(
+            fontWeight: FontWeight.w500,
+            fontFamily: buttonFontFamily,
+          ),
+      110,
+    );
+
+    return Tooltip(
+      message: isOverFlown ? buttonFontFamily : "",
+      margin: const EdgeInsets.only(right: 220),
+      waitDuration: const Duration(milliseconds: 300),
+      child: SizedBox(
+        key: UniqueKey(),
```

Comment:
Do we need a `UniqueKey` here? There is already a `Key` in the `FlowyButton`.

If we do, then try to remove line 144 `key: Key(buttonFontFamily),` and use ValueKey on line 141
```suggestion
        key: ValueKey(buttonFontFamily),
```

Reply:
Checked this.  It looks like neither of the keys is necessary here. The flow is working fine without keys. 

Should I remove both keys or should I pick this suggested change?